src/amgx_cusparse.cu

```diff
@@ -1157,7 +1203,26 @@ inline void Cusparse::bsrmv( cusparseHandle_t handle, cusparseDirection_t dir, c
     }
     else
     {
-        cusparseCheckError(cusparseDbsrmv(handle, dir, trans, mb, nb, nnzb, alpha, descr, bsrVal, bsrRowPtr, bsrColInd, blockDim, x, beta, y));
+        constexpr int block_size = 128;
+        constexpr int warp_size = 32;
+        switch(blockDim) 
+        {
+            case 4:
+                {
+                    const int nblocks = mb / (32) + 1;
+                    Dbsrmv_opt<4, warp_size, block_size><<<nblocks, block_size>>>(mb, *alpha, bsrVal, bsrRowPtr, bsrColInd, x, *beta, y);
+                }
+                break;
+            case 5:
+                {
+                    const int nblocks = mb / (30) + 1;
```

Comment:
I think this should be nblocks = mb / (warps_per_block * rows_per_warp) + 1 = mb / (4 * 6) + 1 = mb / 24 + 1